File: sales_by_match/sbm.js
const count = (array, item) =>{
    let counter = 0;
    for(let i = 0; i < array.length; i++){
        if(array[i] === item){
            counter++;
        }
    }
    return counter;
}

const sock_merchant = (n, ar) => {
    let pairs = 0;
    let uniqueColors = new Set(ar);
    let countSock = {};
    console.log(uniqueColors);
    for (let i = 0; i < uniqueColors.size; i++) {
        countSock[i] = count(ar, uniqueColors);
    }
    console.log(countSock);
    for (const key in countSock) {
        pairs += Math.floor(countSock[key] / 2);
    }
    return pairs;
};


console.log(sock_merchant(9, [10, 20, 20, 10, 10, 30, 50, 10, 20]));

File: sales_by_match/sbm.py
def sock_merchant(n: int, ar:list)->int:
    """
    Function to count the number of matching pairs of socks.
    """
    unique_colors = set(ar)
    count_sock = {sock: ar.count(sock) for sock in unique_colors}
    counter = 0
    for key in count_sock:
        counter += count_sock[key] // 2
    return counter


if __name__ == "__main__":
    print(sock_merchant(9, [10, 20, 20, 10, 10, 30, 50, 10, 20]))